pkg/ossm/ior.go

```diff
@@ -0,0 +1,177 @@
+func TestIOR(t *testing.T) {
+	defer cleanupMultipleIOR()
+	//For 2.4 we need to enable IOR, by default should be disable
+	defer util.RecoverPanic(t)
+	smcpVersion, _ := util.ShellMuteOutput(`oc get smcp/%s -n %s -o jsonpath='{.spec.version}'`, smcpName, meshNamespace)
+	util.Log.Info("SMCP version: ", smcpVersion)
+	iorEnabled, _ := util.ShellMuteOutput(`oc get smcp/%s -n %s -o jsonpath='{.status.appliedValues.istio.gateways.istio-ingressgateway.ior_enabled}'`, smcpName, meshNamespace)
+	util.Log.Info("IOR enabled: ", iorEnabled)
+
+	// Check if IOR is enabled or not by default. Note: for >= 2.4 IOR is disabled by default, so we need to enable it in another subtest. For < 2.4 IOR is enabled by default.
+	t.Run("check_default_ior_state", func(t *testing.T) {
+		defer util.RecoverPanic(t)
+		if strings.Contains(smcpVersion, "v2.4") {
+			util.Log.Info("IOR should be disabled by default")
+			if iorEnabled == "true" {
+				t.Errorf("IOR should be disabled by default")
+			}
+		} else {
+			if iorEnabled == "false" {
+				t.Errorf("IOR should be enabled by default")
+			}
+		}
+	})
+	// Check if we can enable ior for >= 2.4. IOR is disabled by default.
+	t.Run("enable_ior", func(t *testing.T) {
+		defer util.RecoverPanic(t)
+		if strings.Contains(smcpVersion, "v2.4") {
+			util.Log.Info("IOR should be disabled by default")
+			if iorEnabled == "true" {
+				t.Errorf("IOR should be disabled by default")
+			} else {
+				//Need to enable ior on the smcp
+				util.Log.Info("Enabling IOR")
+				util.Shell(`oc patch smcp/%s -n %s --type json -p='[{"op": "add", "path": "/status/appliedValues/istio/gateways/istio-ingressgateway/ior_enabled", "value": true}]'`, smcpName, meshNamespace)
+				_, err := util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName)
+				if err != nil {
+					t.Fatal("SMCP is not ready")
+				}
+				iorEnabled, _ := util.ShellMuteOutput(`oc get smcp/%s -n %s -o jsonpath='{.status.appliedValues.istio.gateways.istio-ingressgateway.ior_enabled}'`, smcpName, meshNamespace)
+				util.Log.Info("IOR enabled: ", iorEnabled)
+				if iorEnabled == "false" {
+					t.Errorf("IOR should be enabled")
+				}
+			}
+		} else {
+			util.Log.Info("IOR should be enabled by default")
+			if iorEnabled == "false" {
+				t.Errorf("IOR should be enabled by default")
+			} else {
+				t.Skip("IOR is enabled by default, no need to enable it. Skipping this test")
+			}
+		}
+	})
+	util.Log.Info("Setup IOR")
+	util.Log.Info("Create 50 new namespaces")
+	util.Shell(`../scripts/smmr/create_members_50.sh`)
+	util.Log.Info("Namespaces and smmr created...")
+	util.Log.Info("Create gateway in each namespace")
+	util.Shell(`../scripts/gateway/create_multiple_gateway.sh`)
+	util.Log.Info("Gateways created...")
+	routes, _ := util.ShellMuteOutput(`oc get -n %s route -o jsonpath='{.items}'`, meshNamespace)
+	routesData := ParseRoutes(routes)
```

Comment:
I think it's better for the regression test to check that non routes are deleted so for me yes. If not What can be the criteria? Because we will have 50 new routes, we should only verify how many?

Reply:
OK, I understand the concern. So, what do you think is the best way, we can pick 2 o 4 routes randomly and check if those routes exist and have the same time in the routes after the changes.

Reply:
I think we can just make sure we got the expected number of routes. In this case, we create 100 gateways and each with one host. There should be 100 routes created.